pkg/kubectl/cmd/edit.go

```diff
@@ -245,40 +245,36 @@ func runEdit(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args
 			}
 
 			// parse the edited file
-			updates, err := resourceMapper.InfoForData(edited, "edited-file")
-			if err != nil {
+			var updatedVisitor resource.Visitor
+			if updatedInfos, err := updatedResultsGetter(edited).Infos(); err != nil {
 				// syntax error
 				containsError = true
 				results.header.reasons = append(results.header.reasons, editReason{head: fmt.Sprintf("The edited file had a syntax error: %v", err)})
 				continue
+			} else {
```

Comment:
Ug I see now, it is because of the scoping of `updatedInfos`.  FWIW, the way the code is structured makes me do a double take to figure out if updatedVisitor is guaranteed to always be set.